### PR TITLE
Add the support of profiling and multi-GPU running 

### LINKED ISSUE #622: Multi-device support for model training

### PR DESCRIPTION
**Pull request by kuke**
Resolve #622

### CODE REVIEW

**Review by zhxfl — DISMISSED** [review]

File: fluid/DeepASR/train.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+    exe = fluid.Executor(place)
+    exe.run(fluid.default_startup_program())
+
+    ltrans = [
+        trans_add_delta.TransAddDelta(2, 2),
+        trans_mean_variance_norm.TransMeanVarianceNorm(args.mean_var),
+        trans_splice.TransSplice()
+    ]
+
+    res_feature = fluid.LoDTensor()
```
Comment (pkuyym) [inline fluid/DeepASR/train.py]: Please think a better name.

File: fluid/DeepASR/train.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+        # test data reader
+        test_data_reader = reader.DataReader(args.val_feature_lst,
+                                             args.val_label_lst)
+        test_data_reader.set_transformers(ltrans)
+        test_costs, test_accs = [], []
+        for batch_id, batch_data in enumerate(
+                test_data_reader.batch_iterator(args.batch_size,
+                                                args.minimum_batch_size)):
+            # load_data
+            (bat_feature, bat_label, lod) = batch_data
```
Comment (pkuyym) [inline fluid/DeepASR/train.py]: Same as above

**Review by pkuyym — CHANGES_REQUESTED** [review]

File: fluid/DeepASR/train.py (hunk 1 above)
Comment (kuke) [reply fluid/DeepASR/train.py]: Done

File: fluid/DeepASR/train.py (hunk 2 above)
Comment (kuke) [reply fluid/DeepASR/train.py]: Done

**Review by kuke — COMMENTED** [review]
All done, thx

**Comment by zhxfl** [comment]
LGTM

File: fluid/DeepASR/model.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+import paddle.v2 as paddle
+import paddle.v2.fluid as fluid
+
+
+def stacked_lstmp_model(hidden_dim,
+                        proj_dim,
+                        stacked_num,
+                        parallel=False,
+                        is_train=True,
+                        class_num=1749):
```
Comment (pkuyym) [inline fluid/DeepASR/model.py]: I think we should not give a default value for `class_num`

File: fluid/DeepASR/model.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+	parallel(bool): Run in parallel or not, default `False`.
+	is_train(bool): Run in training phase or not, default `True`.
+	class_dim(int): The number of output classes.
+    """
+
+    # network configuration
+    def _net_conf(feature, label):
+        seq_conv1 = fluid.layers.sequence_conv(
+            input=feature,
+            num_filters=1024,
```
Comment (pkuyym) [inline fluid/DeepASR/model.py]: We can make this configurable later, just mark here.

File: fluid/DeepASR/model.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+                momentum=0.9,
+                epsilon=1e-05,
+                data_layout='NCHW')
+            stack_input = bn
+
+        prediction = fluid.layers.fc(input=stack_input,
+                                     size=class_num,
+                                     act='softmax')
+
+        cost = fluid.layers.cross_entropy(input=prediction, label=label)
```
Comment (pkuyym) [inline fluid/DeepASR/model.py]: If `is_traing==False`, `cost` is unnecessary, since there maybe no label.

**Review by pkuyym — CHANGES_REQUESTED** [review]

**Comment by pkuyym** [comment]
Please consider reorganize the directory structure, for example:
```
model_utils/model.py
train.py
utils/profile.py
test.py
```

File: fluid/DeepASR/model.py (hunk 3 above)
Comment (kuke) [reply fluid/DeepASR/model.py]: Done

File: fluid/DeepASR/model.py (hunk 5 above)
Comment (kuke) [reply fluid/DeepASR/model.py]: ```cost``` and ```acc``` will be pruned when saving inference model

**Review by kuke — COMMENTED** [review]
Updated. Thx

**Review by pkuyym — APPROVED** [review]
LGTM

**Pull request merged**